cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go

```diff
@@ -235,190 +235,198 @@ func (t Template) ControllersPolicy() *infrav1.PolicyDocument {
 			})
 		}
 	}
-	if !t.Spec.EKS.Disable {
-		allowedIAMActions := infrav1.Actions{
-			"iam:GetRole",
-			"iam:ListAttachedRolePolicies",
-		}
+	if t.Spec.EventBridge.Enable {
 		statement = append(statement, infrav1.StatementEntry{
-			Effect: infrav1.EffectAllow,
-			Resource: infrav1.Resources{
-				"arn:*:ssm:*:*:parameter/aws/service/eks/optimized-ami/*",
-			},
+			Effect:   infrav1.EffectAllow,
+			Resource: infrav1.Resources{infrav1.Any},
 			Action: infrav1.Actions{
-				"ssm:GetParameter",
+				"events:DeleteRule",
+				"events:DescribeRule",
+				"events:ListTargetsByRule",
+				"events:PutRule",
+				"events:PutTargets",
+				"events:RemoveTargets",
+				"sqs:CreateQueue",
+				"sqs:DeleteMessage",
+				"sqs:DeleteQueue",
+				"sqs:GetQueueAttributes",
+				"sqs:GetQueueUrl",
+				"sqs:ReceiveMessage",
+				"sqs:SetQueueAttributes",
 			},
 		})
+	}
+
+	return &infrav1.PolicyDocument{
+		Version:   infrav1.CurrentVersion,
+		Statement: statement,
+	}
+}
+
+// ControllersPolicyEKS creates a policy from a template for AWS Controllers.
```

Comment:
@johananl I think it would be nice to document the 10 policies limit somewhere in here, to indicate why other resources has not been put in separate policies. We could as well do this in #2271 though.

Reply:
I think it's ok for now. The 10 policies limit should be relatively well understood by folk in the IAM space.